src/lib/storage/encoding_type.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <boost/hana/at_key.hpp>
+#include <boost/hana/contains.hpp>
+#include <boost/hana/equal.hpp>
+#include <boost/hana/map.hpp>
+#include <boost/hana/pair.hpp>
+#include <boost/hana/type.hpp>
+
+#include <cstdint>
+
+#include "all_type_variant.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+namespace hana = boost::hana;
+
+enum class EncodingType : uint8_t { Unencoded, DeprecatedDictionary, Dictionary, RunLength };
+
+namespace detail {
+
+// Use all_data_types if the encoding supports all data types.
```

Comment:
We could also move this comment down and use `data_types` if the encoding supports all data types.

Reply:
done